### PR TITLE
Check controlPlaneTopology in infrastructures.config for hypershift-hosted

### PR DESCRIPTION
**Pull request by liangxia**
Based on https://github.com/openshift/release/pull/49618#issuecomment-1988245996

### CODE REVIEW

**Comment by liangxia** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-guest-critical-f7 periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-azure-ipi-nat-gateway-tp-arm-f14

**Comment by liangxia** [comment]
Installation failures not related to changes in this PR. Retest,

/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-aws-ipi-ovn-hypershift-guest-critical-f7 periodic-ci-openshift-openshift-tests-private-release-4.15-multi-nightly-azure-ipi-nat-gateway-tp-arm-f14

File: ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh [hunk 1]
```diff
@@ -130,9 +130,9 @@ function filter_test_by_proxy() {
     echo_e2e_tags
 }
 function filter_test_by_hypershift() {
-    local masterno
-    masterno="$(oc get nodes --no-headers -l node-role.kubernetes.io/master= | wc -l)"
-    if [[ $masterno -eq 0 ]] ; then
+    local topo
+    topo="$(oc get infrastructures.config.openshift.io cluster -o yaml | yq '.status.controlPlaneTopology')"
+    if [[ -n "$topo" ]] && [[ "$topo" = 'External' ]] ; then
```
Comment (JianLi-RH) [inline ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh:135]: Do we really need 2 criteria here? I think [[ "$topo" = 'External' ]] is enough

Comment (liangxia) [reply ci-operator/step-registry/cucushift/e2e/cucushift-e2e-commands.sh]: Done

**Pull request merged**